core/state_processor.go

```diff
@@ -109,17 +115,21 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 				return nil, nil, 0, err
 			}
 
-			// move N=500 accounts into the verkle tree, starting with the
+			const maxMovedCount = 500
+			// mkv will be assiting in the collection of up to maxMovedCount key values to be migrated to the VKT.
+			// It has internal caches to do efficient MPT->VKT key calculations, which will be discarded after
+			// this function.
+			mkv := &keyValueMigrator{}
```

Comment:
This `keyValueMigrator` is a component that encapsulates a bit capturing the relevant walked key-values. I did this to keep the walk clean and let the component do the work. I'll comment on it later.